web-ui/src/main/resources/catalog/components/common/map/esriService.js

```diff
@@ -160,6 +160,38 @@
             }
           }
           return [width, height];
+        },
+
+        /**
+         * Returns a promise resolving to the capabilities document of an ESRI Rest service.
+         * @param {String} url
+         * @return {Promise<String>} capabilities document
+         */
+        getCapabilities: function(url) {
+          var timeout = 60 * 1000;
+          var defer = $q.defer();
+          if (url) {
+            url = url + '?f=json';
+
+            if (gnUrlUtils.isValid(url)) {
```

Comment:
If the URL is not valid the returned promise will never resolve nor reject; I think you can remove this check and let `$http` fail like it should on a malformed URL.

Reply:
ok, will update.